alacritty_terminal/src/term/mod.rs

```diff
@@ -1486,10 +1486,10 @@ impl<T: EventListener> Handler for Term<T> {
         if self.mode.contains(TermMode::INSERT) && self.grid.cursor.point.col + width < num_cols {
             let line = self.grid.cursor.point.line;
             let col = self.grid.cursor.point.col;
-            let line = &mut self.grid[line];
+            let line = &mut self.grid[line][..];
 
-            for col in IndexRange::from(col..(num_cols - width)).rev() {
-                line[col + width] = line[col].clone();
+            for col in (col.0..(num_cols - width).0).rev() {
+                line.swap(col + width, col);
             }
```

Comment:
Honestly I kinda like naming the line number `line` while naming the row itself `row`. I think that separates the two a bit nicer for these instances where both are present. This applies to all changes.

Shadowing the line number with the line slice just seems a bit weird to me?